src/components/Table/useColumns.ts

```diff
@@ -0,0 +1,21 @@
+import { Dispatch, SetStateAction, useMemo, useState } from "react";
+import { GridColumn, Kinded } from "./GridTable";
+
+export interface Columns<R extends Kinded, S = {}> {
```

Comment:
Can we move this `type` to the bottom of the file. Otherwise people may assume that it is the interface for the hook, which it is not. This is per our component styling guide: https://homebound.atlassian.net/wiki/spaces/ENG/pages/339574863/Component+Style+Guide#Component-Structure